### ISSUE TITLE
`warning: previous definition of as_json was here` against Ruby 3.4.0dev 

### ISSUE THREAD

**Issue by yahonda** [rails/rails#53391]
Managed to reproduce Rails Nightly CI failure at https://buildkite.com/rails/rails-nightly/builds/1184#0192a695-b863-48d1-aa4b-c17086e58223 since https://github.com/ruby/ruby/commit/eb8cf1d60e4436bfebf52fbf4342ee6e5e18ba05

### Steps to reproduce

```ruby
git clone https://github.com/rails/rails
cd rails
bundle install
cd activesupport
RAILS_STRICT_WARNINGS=1 bin/test
```

### Expected behavior
It should pass.

### Actual behavior
It raises `warning: previous definition of as_json was here` warning and fails since `RAILS_STRICT_WARNINGS=1` is set.

```ruby
$ cd activesupport
$ RAILS_STRICT_WARNINGS=1 bin/test
/home/yahonda/.rbenv/versions/trunk/lib/ruby/gems/3.4.0+0/gems/minitest-5.25.1/lib/minitest/mock.rb:33: warning: redefining 'object_id' may cause serious problems
/home/yahonda/src/github.com/rails/rails/activesupport/lib/active_support/core_ext/object/json.rb:243: warning: method redefined; discarding old as_json
/home/yahonda/.rbenv/versions/trunk/lib/ruby/3.4.0+0/ipaddr.rb:231: warning: previous definition of as_json was here
/home/yahonda/src/github.com/rails/rails/activesupport/lib/active_support/testing/strict_warnings.rb:38:in 'ActiveSupport::RaiseWarnings#warn': /home/yahonda/src/github.com/rails/rails/activesupport/lib/active_support/core_ext/object/json.rb:243: warning: method redefined; discarding old as_json (ActiveSupport::RaiseWarnings::WarningError)
/home/yahonda/.rbenv/versions/trunk/lib/ruby/3.4.0+0/ipaddr.rb:231: warning: previous definition of as_json was here

	from /home/yahonda/src/github.com/rails/rails/activesupport/lib/active_support/core_ext/object/json.rb:243:in '<class:IPAddr>'
	from /home/yahonda/src/github.com/rails/rails/activesupport/lib/active_support/core_ext/object/json.rb:242:in '<top (required)>'
	from /home/yahonda/.rbenv/versions/trunk/lib/ruby/3.4.0+0/bundled_gems.rb:76:in 'Kernel.require'
	from /home/yahonda/.rbenv/versions/trunk/lib/ruby/3.4.0+0/bundled_gems.rb:76:in 'block (2 levels) in Kernel#replace_require'
	from /home/yahonda/src/github.com/rails/rails/activesupport/lib/active_support/json/encoding.rb:3:in '<top (required)>'
	from /home/yahonda/.rbenv/versions/trunk/lib/ruby/3.4.0+0/bundled_gems.rb:76:in 'Kernel.require'
	from /home/yahonda/.rbenv/versions/trunk/lib/ruby/3.4.0+0/bundled_gems.rb:76:in 'block (2 levels) in Kernel#replace_require'
	from /home/yahonda/src/github.com/rails/rails/activesupport/lib/active_support/json.rb:4:in '<top (required)>'
	from /home/yahonda/.rbenv/versions/trunk/lib/ruby/3.4.0+0/bundled_gems.rb:76:in 'Kernel.require'
	from /home/yahonda/.rbenv/versions/trunk/lib/ruby/3.4.0+0/bundled_gems.rb:76:in 'block (2 levels) in Kernel#replace_require'
	from /home/yahonda/src/github.com/rails/rails/activesupport/test/core_ext/duration_test.rb:6:in '<top (required)>'
	from /home/yahonda/.rbenv/versions/trunk/lib/ruby/3.4.0+0/bundled_gems.rb:76:in 'Kernel.require'
	from /home/yahonda/.rbenv/versions/trunk/lib/ruby/3.4.0+0/bundled_gems.rb:76:in 'block (2 levels) in Kernel#replace_require'
	from /home/yahonda/src/github.com/rails/rails/railties/lib/rails/test_unit/runner.rb:62:in 'block in Rails::TestUnit::Runner.load_tests'
	from <internal:array>:42:in 'Array#each'
	from /home/yahonda/src/github.com/rails/rails/railties/lib/rails/test_unit/runner.rb:60:in 'Rails::TestUnit::Runner.load_tests'
	from /home/yahonda/src/github.com/rails/rails/railties/lib/rails/test_unit/runner.rb:52:in 'Rails::TestUnit::Runner.run'
	from /home/yahonda/src/github.com/rails/rails/tools/test.rb:18:in '<top (required)>'
	from bin/test:5:in 'Kernel#require_relative'
	from bin/test:5:in '<main>'
$
```

### System configuration
**Rails version**: main branch

**Ruby version**: ruby 3.4.0dev (2024-10-19T13:00:02Z master 1c653cabfc) +PRISM [x86_64-linux]

**Comment by yahonda** [comment]
Here is the original pull request for ruby/ipaddr https://github.com/ruby/ipaddr/pull/77

**Comment by yahonda** [comment]
https://github.com/rails/rails/commit/e7d207f02fe8bf6e351de7da106839b0ab14bbea added `IPAddr#as_json` to Rails.

**Comment by taketo1113** [comment]
@yahonda I created PR (#53393) to fix another issue.
It will fix this warning.

**Comment by yahonda** [comment]
Closed via #53393
